src/sentry/api/serializers/models/alert_rule.py

```diff
@@ -1,5 +1,5 @@
 from collections import defaultdict
-from typing import MutableMapping
+from typing import Any, DefaultDict, Dict, List, Mapping, MutableMapping, Optional, Sequence, Union
```

Comment:
if you `from __future__ import annotations` you'll be able to use the nice `list[...]` / `dict[...]` `| None` syntax instead of Optional / Union / Dict / List / etc.

Reply:
As someone who doesn't do this much, how would I know based on the [docs](https://docs.python.org/3/library/typing.html#typing.Optional) that I'm supposed to use one over the other?

Reply:
we ~mostly follow the recommendations here https://typing.readthedocs.io/en/latest/source/best_practices.html